src/ast.rs

```diff
@@ -4,7 +4,9 @@ mod untyped;
 pub use self::typed::TypedExpr;
 pub use self::untyped::UntypedExpr;
 
-use crate::typ::{self, ModuleValueConstructor, PatternConstructor, Type, ValueConstructor};
+use crate::typ::{
+    self, bool, int, ModuleValueConstructor, PatternConstructor, Type, ValueConstructor,
```

Comment:
Small style nitpick: Could we not import `bool` and `int` and instead in the function call `typ::int()` and `typ::bool()`? It makes it a little clearer where the function is defined (as the name is pretty generic).